product-service/src/functions/catalogBatchProcess/handler.ts

```diff
@@ -0,0 +1,58 @@
+import {SNSClient} from "@aws-sdk/client-sns";
+import {PublishCommand} from '@aws-sdk/client-sns';
+import {createProduct} from "../createProduct/handler"
+import {handleHttpError} from '@libs/lambda';
+import {formatJSONResponse} from "@libs/api-gateway";
+
+const catalogBatchProcess = async event => {
+
+    const snsClient = new SNSClient({ region: "us-east-1" });
+
+    try {
+        const products: {title, description, price, count}[] = event.Records.map(({ body }) => JSON.parse(body));;
+
+        for (const product of products) {
+            const { title, description, price, count } = product;
+
+            if (!Boolean(title && description && price && count)) {
+                return
+            }
+
+            if (await createProduct({
```

Comment:
i think it would be better to move the create product method to a separate service than invoking api event handler, it's a bad practice